tests/test_mlkit.py

```diff
@@ -325,6 +329,17 @@ def instrument_mlkit_service(status=200, payload=None, operations=False, app=Non
             session_url, adapter(payload, status, recorder))
     return recorder
 
+class _TestStorageClient(object):
+    @staticmethod
+    def upload(bucket_name, model_file_name, app):
```

Comment:
Just a suggestion. Feel free to ignore if it's not important. You can record the arguments in some data structure (e.g. list of tuples) and then assert for them.

```
def __init__(self):
  self.upload_args = []

def upload(...):
  self.upload_args.append((bucket_name, model_file_name, app))
  ...

# And later in the tests
args = test_client.upload_args[0]
asserts args[0] == expected_bucket_name
```

Reply:
We already have the bucket_name and model_file_name in the returned uri for upload. 
I changed the sign_uri to also use the param.